grakn-test/src/test/java/ai/grakn/generator/TaskStates.java

```diff
@@ -79,7 +80,12 @@ public TaskState generate(SourceOfRandomness random, GenerationStatus status) {
         Json configuration = Json.object();
         TaskState taskState = TaskState.of(taskClass, creator, TaskSchedule.now(), configuration, taskId);
         configuration.set("id", taskState.getId().getValue());
-        return taskState.status(taskStatus);
+        if(taskStatus == TaskStatus.RUNNING){
+            taskState.setRunning(EngineID.me());
+        } else {
+            taskState.status(taskStatus);
```

Comment:
Only `RUNNING` tasks should have an EngineID because this is used during failover. When an engine fails, we ask the storage for any `RUNNING` tasks that may have been on that machine so that we can resubmit them to another machine. 

We could potentially keep the EngineID and then filter by `RUNNING` state, but it would make querying during failover much heavier.